### PR TITLE
Add multi-qubit and multi-control gates.

### PR DESCRIPTION
**Pull request by sergeisakov**

### CODE REVIEW

File: tests/simulator_testfixture.h [hunk 1]
```diff
@@ -46,12 +48,19 @@ void TestApplyGate1() {
   auto gate5 = GateZ<fp_type>::Create(4, 0);
   auto gate6 = GateHZ2<fp_type>::Create(5, 0);
 
+  puts("obs1");
```
Comment (95-martin-orion) [inline tests/simulator_testfixture.h:51]: Remove debug lines?

File: lib/simulator_avx.h [hunk 2]
```diff
@@ -19,15 +19,15 @@
...
 
+#include "bits.h"
 #include "statespace_avx.h"
 
 namespace qsim {
 
 /**
- * Quantum circuit simulator with AVX vectorization.
- */
+* Quantum circuit simulator with AVX vectorization.
```
Comment (95-martin-orion) [inline lib/simulator_avx.h:29]: Top-level comment: if this is generated code, could you include the script used to generate it in the PR? That would be easier to review, and provides a reference for how these changes were made.

**Review by 95-martin-orion — CHANGES_REQUESTED** [review]

File: tests/simulator_testfixture.h (hunk 1 above)
Comment (sergeisakov) [reply tests/simulator_testfixture.h]: Done.

File: lib/simulator_avx.h (hunk 2 above)
Comment (sergeisakov) [reply lib/simulator_avx.h]: I would prefer to not include the script at this point.

**Review by 95-martin-orion — APPROVED** [review]

File: lib/simulator_avx.h (hunk 2 above)
Comment (95-martin-orion) [reply lib/simulator_avx.h]: Discussed offline: submitting the script to the repo is lower priority. #227 has been opened to track that work.

**Pull request merged**